# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/Frontend2.cpp
# Commit: 8af30c6d9bb9
# Full Hash: 8af30c6d9bb91a1ed9a5bee712252bbe1bd20d85
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2020-11-14 09:46:25
# Regressor Bug: 1675241
# File Overlap Count: 1
# Description:
#   Bug 1675241 - Part 11: Use TaggedParserAtomIndex in ScriptThingVariant. r=tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96918
# ==============================================================================

diff -r cb0e58368cd4 -r 8af30c6d9bb9 js/src/frontend/Frontend2.cpp
--- a/js/src/frontend/Frontend2.cpp	Sat Nov 14 00:03:28 2020 +0000
+++ b/js/src/frontend/Frontend2.cpp	Sat Nov 14 00:03:36 2020 +0000
@@ -380,7 +380,7 @@
         break;
       }
       case SmooshGCThing::Tag::Atom: {
-        new (raw) ScriptThingVariant(allAtoms[item.AsAtom()]);
+        new (raw) ScriptThingVariant(allAtoms[item.AsAtom()]->toIndex());
         break;
       }
       case SmooshGCThing::Tag::Function: {